frontend/src/screens/GroupsScreen.tsx

```diff
@@ -1,16 +1,95 @@
-import { Text } from "react-native";
+import { H1, H2, P } from "@expo/html-elements";
+import { Link } from "@react-navigation/native";
+import { StyleSheet, View } from "react-native";
+import reactStringReplace from "react-string-replace";
 
-import { useGetGroups } from "../api/apiComponents";
-import ScreenWrapper from "../components/ScreenWrapper";
+import { theme } from "@/config/theme";
+import { useResponsiveLayout } from "@/hooks/useResponsiveLayout";
+import { useTranslation } from "@/hooks/useTranslation";
+import { useTypographyStyles } from "@/hooks/useTypographyStyles";
+import { useGetGroups } from "@/api/apiComponents";
+import { Group } from "@/api/apiSchemas";
+import { GroupList } from "@/components/GroupCard";
+import ScreenWrapper from "@/components/ScreenWrapper";
+
+export const GroupsScreen = () => {
+  const { data: { data: groups } = {}, isLoading, isError } = useGetGroups({});
+  const { t } = useTranslation();
+  const { h1, h2, bodyNormal, link } = useTypographyStyles();
+  const styles = useStyles();
+
+  const groupTypes: NonNullable<Group["Type"]>[] = [
```

Comment:
Could we get this from the database?

Reply:
it's a typescript type, so no, it only works during compile time. It's being retrieved from the types file that is generated from the API spec

Reply:
Gotcha!